linter/linter.go

```diff
@@ -43,6 +44,16 @@ type Result struct {
 	Schemas       map[string]*tengo.Schema // Keyed by dir path and optionally schema name
 }
 
+// SortByFile implements the sort.Interface for []*Annotation based on
+// the Annotation.Statement.File field.
+type SortByFile []*Annotation
+
+func (a SortByFile) Len() int      { return len(a) }
+func (a SortByFile) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a SortByFile) Less(i, j int) bool {
+	return a[i].Statement.File < a[j].Statement.File
```

Comment:
One other minor thing I forgot to mention -- probably better to name the new type `sortByFile` (non-exported) since it's internal implementation details / not needed outside of the package.

Reply:
Thanks. Good points, I've fixed these in the latest commit.